webapi/Controllers/ChatHistoryController.cs

```diff
@@ -240,4 +241,100 @@ public async Task<IActionResult> GetAllChatSessionsAsync(string userId)
 
         return this.NotFound($"No chat session found for chat id '{chatId}'.");
     }
+
+    /// <summary>
+    /// Delete a chat session.
+    /// </summary>
+    /// <param name="userId">Unique Id of user who initiated the request.</param>
+    /// <param name="sessionId">The chat id.</param>
+    [HttpDelete]
+    [Route("chatSession/{sessionId:guid}")]
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> DeleteChatSessionAsync([FromServices] IHubContext<MessageRelayHub> messageRelayHubContext, [FromBody] DeleteChatRequest requestBody, Guid sessionId)
+    {
+        var chatId = sessionId.ToString();
+        var userId = requestBody.UserId;
+
+        if (userId == null)
+        {
+            return this.BadRequest("UserId cannot be null.");
+        }
+
+        ChatSession? chatToDelete = null;
+        try
+        {
+            // Make sure the chat session exists
+            chatToDelete = await this._sessionRepository.FindByIdAsync(chatId);
+        }
+        catch (KeyNotFoundException)
+        {
+            return this.NotFound($"No chat session found for chat id '{chatId}'.");
+        }
+
+        // Delete any resources associated with the chat session.
+        var deleteResourcesResult = await this.DeleteChatResourcesAsync(messageRelayHubContext, sessionId) as StatusCodeResult;
+        if (deleteResourcesResult?.StatusCode != 204)
+        {
+            return this.StatusCode(424, $"Failed to delete resources for chat id '{chatId}'.");
+        }
+
+        // Delete chat session and broadcast update to all participants.
+        await this._sessionRepository.DeleteAsync(chatToDelete);
+        await messageRelayHubContext.Clients.Group(chatId).SendAsync(ChatDeletedClientCall, chatId, userId);
+
+        return this.NoContent();
+    }
+
+    /// <summary>
+    /// Deletes all associated resources (messages, memories, participants) associated with a chat session.
+    /// </summary>
+    /// <param name="sessionId">The chat id.</param>
+    [HttpDelete]
+    [Route("chatSession/{sessionId:guid}/resources")]
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
```

Comment:
Oh yes, I was calling this from the webapp but since I frontloaded the resource deletion, we can retry chat deletion altogether on failure and this is no longer needed.

Reply:
Removed